identify/identify.py

```diff
@@ -203,7 +216,14 @@ def parse_shebang_from_file(path: str) -> Tuple[str, ...]:
     """Parse the shebang given a file path."""
     if not os.path.lexists(path):
         raise ValueError(f'{path} does not exist.')
-    if not os.access(path, os.X_OK):
+    ext = os.path.splitext(os.path.split(path)[-1])[-1].lstrip('.')
+    if (
+        ext not in extensions.EXTENSIONS_NEED_INTERPRETER_CHECK and
+        not os.access(
+            path,
+            os.X_OK,
+        )
+    ):
         return ()
 
```

Comment:
```suggestion
```

@asottile The only function that calls this already checks this same condition before running `parse_shebang_from_file()`, so is there a reason it needs to be checked twice? There might be something I'm missing, but if not, eliminating the redundant check would reduce complexity and increase performance without an obvious downside, and it also prevents any other third-party consumers from using it on files not marked executable if that restriction doesn't make sense for their application.